scrapper.py

```diff
@@ -26,38 +37,122 @@ class Crawler:
     Crawler implementation
     """
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        link_bs = article_bs.find('a')
+        return ROOT_URL + link_bs['href']
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for seed_url in self.seed_urls:
+            if len(self.urls) >= self.max_articles:
+                break
+            response = requests.get(seed_url, HEADERS)
+            time.sleep(1)
+            article_bs = BeautifulSoup(response.text, 'html.parser')
+            self._extract_url(article_bs)
+            class_bs = article_bs.find('div', class_='view-content view-rows')
+            title_bs = class_bs.find_all('td', class_="views-field views-field-title table__cell")
+            for link_bs in title_bs:
+                if len(self.urls) >= self.max_articles:
+                    break
+                self.urls.append(self._extract_url(link_bs))
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    path = Path(base_path)
+    if path.is_dir():
+        shutil.rmtree(path)
+    path.mkdir(parents=True, exist_ok=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path, 'r', encoding='utf-8') as file:
+        config = json.load(file)
+    seed_urls = config['seed_urls']
```

Comment:
are you sure those keys will be always present in the configuration file? would you consider such incomplete config as a correct one?

Reply:
Done